src/Installer.php

```diff
@@ -255,6 +321,44 @@ private function downloadFile(HttpClient $httpClient, $url, $targetPath)
     }
 
     /**
+     * Downloads the available puli versions.
+     *
+     * @param HttpClient $httpClient The client to use.
+     * @param string     $url        The URL to download.
+     *
+     * @return array The available versions, null if the download failed.
+     *
+     * @throws RuntimeException If an error occurs.
+     */
+    public function downloadVersions(HttpClient $httpClient, $url)
+    {
+        ErrorHandler::register();
+        $versions = $httpClient->download($url);
+        ErrorHandler::unregister();
+
+        if (ErrorHandler::hasErrors()) {
+            throw new RuntimeException(sprintf(
+                "Could not download %s:\n%s",
+                $url,
+                implode("\n", ErrorHandler::getErrors())
+            ));
+        }
+
+        $versions = json_decode($versions);
+        if (json_last_error() !== JSON_ERROR_NONE || !is_array($versions)) {
+            throw new RuntimeException(sprintf(
+                "Could not download %s:\n%s",
+                $url,
+                'Malformed JSON returned.'
```

Comment:
Can you do this change? Also, everywhere in the installer where we use "\n" we should actually use PHP_EOL.